processor/tailsamplingprocessor/sampling/string_tag_filter.go

```diff
@@ -15,31 +15,37 @@
 package sampling
 
 import (
+	"regexp"
+
+	"github.com/golang/groupcache/lru"
```

Comment:
Maybe use https://github.com/hashicorp/golang-lru instead of https://github.com/golang/groupcache?

Reply:
Tested `hashicorp/lru` has worse performance than `google/lru`. and the latter one has been used in the other components in OTel.